jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/FilesTask.java

```diff
@@ -0,0 +1,140 @@
+   * @param projectDependencyJars the set of jar files associated with each project dependency
+   *     encountered
+   */
+  private static void listFilesForProject(Project project, Set<File> projectDependencyJars) {
+    JavaPluginConvention javaConvention =
+        project.getConvention().getPlugin(JavaPluginConvention.class);
+    SourceSet mainSourceSet = javaConvention.getSourceSets().findByName("main");
+    if (mainSourceSet == null) {
+      return;
+    }
+
+    // Print build.gradle
+    System.out.println(project.getBuildFile());
+
+    // Print settings.gradle
+    Path settingsFile = project.getProjectDir().toPath().resolve(Settings.DEFAULT_SETTINGS_FILE);
+    if (Files.exists(settingsFile)) {
+      System.out.println(settingsFile);
+    }
+
+    // Print sources
+    mainSourceSet.getAllJava().getSourceDirectories().forEach(System.out::println);
+
+    // Print resources
+    mainSourceSet.getResources().getSourceDirectories().forEach(System.out::println);
+
+    // Find project dependencies
+    for (Configuration configuration :
+        project.getConfigurations().getByName("runtime").getHierarchy()) {
+      for (Dependency dependency : configuration.getDependencies()) {
+        if (dependency instanceof ProjectDependency) {
+          // Keep track of project dependency jars
+          ProjectDependency projectDependency = (ProjectDependency) dependency;
+          String configurationName = projectDependency.getTargetConfiguration();
+          if (configurationName == null) {
+            configurationName = "default";
+          }
+          Project dependencyProject = projectDependency.getDependencyProject();
+          for (Configuration targetConfiguration :
+              dependencyProject.getConfigurations().getByName(configurationName).getHierarchy()) {
+            for (PublishArtifact artifact : targetConfiguration.getArtifacts()) {
+              projectDependencyJars.add(artifact.getFile());
+            }
+          }
+
+          // Print project dependency's build/source/resource files
+          listFilesForProject(dependencyProject, projectDependencyJars);
+        }
+      }
+    }
+  }
+
+  @Nullable private JibExtension jibExtension;
+
+  public FilesTask setJibExtension(JibExtension jibExtension) {
+    this.jibExtension = jibExtension;
+    return this;
+  }
+
+  @TaskAction
+  public void listFiles() {
+    Preconditions.checkNotNull(jibExtension);
+    Project project = getProject();
+
+    // If this is not the root project, print the root project's build.gradle and settings.gradle
+    if (project != project.getRootProject()) {
+      System.out.println(project.getRootProject().getBuildFile());
+      Path settingsFiles = project.getRootDir().toPath().resolve(Settings.DEFAULT_SETTINGS_FILE);
+      if (Files.exists(settingsFiles)) {
+        System.out.println(settingsFiles);
+      }
+    }
+
+    // Print extra layer
+    if (project.getPlugins().hasPlugin(JibPlugin.class)) {
```

Comment:
What is the reasoning for checking if there is `JibPlugin`?

Reply:
The mojo does this, I'm not sure if it's necessary here.

Reply:
That was because maven is weird and needed to run the goal directly on the other projects (we couldn't access enough information from the project under consideration). And we needed to make sure non-jib modules didn't give us jib data.

But with gradle just lets us go inspect all the sub-projects, so it shouldn't be necessary here.